app/script/auth/module/action/ValidationError.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+export default class ValidationError extends Error {
+  constructor(params) {
+    super();
+    this.name = this.constructor.name;
+    this.label = params.label;
+    this.message = params.label;
+  }
+
+  is = label => {
+    return this.label === label;
+  };
+
+  static getAllPropertyNames(obj) {
+    let props = [];
+    do {
+      props = props.concat(Object.getOwnPropertyNames(obj));
+    } while ((obj = Object.getPrototypeOf(obj)));
+    return props;
+  }
+
+  static handleValidationState(fieldName, validationState) {
+    const field = ValidationError.getFieldByName(fieldName);
+    const validationStateKeys = ValidationError.getAllPropertyNames(validationState);
+    const errorKeys = [];
+    Object.keys(ValidationError.ERROR).forEach(errorName => errorKeys.push(ValidationError.ERROR[errorName]));
+    for (const key of validationStateKeys) {
+      if (errorKeys.includes(key) && validationState[key]) {
+        return new ValidationError({
+          label: field[ValidationError.getErrorByKey(key)],
+        });
```

Comment:
This looks like you can just to a `map` on `Object.keys` or with the help of babel even on `Object.entries()` to full `errorKeys` immeditately without all these validations.